File: jdroid-android-core/src/main/java/com/jdroid/android/navdrawer/NavDrawerItem.java
package com.jdroid.android.navdrawer;

import android.app.Activity;
import androidx.annotation.Nullable;

import com.jdroid.android.activity.AbstractFragmentActivity;

public interface NavDrawerItem {

	public Integer getItemId();

	public void startActivity(@Nullable AbstractFragmentActivity currentActivity);

	public Boolean matchesActivity(Activity activity);

	public Class<? extends Activity> getActivityClass();

}

File: jdroid-android-core/src/main/java/com/jdroid/android/navdrawer/NavDrawerItem.kt
package com.jdroid.android.navdrawer

import android.app.Activity

import com.jdroid.android.activity.AbstractFragmentActivity

interface NavDrawerItem {

    fun getItemId(): Int

    fun startActivity(currentActivity: AbstractFragmentActivity?)

    fun matchesActivity(activity: Activity): Boolean

    fun getActivityClass(): Class<out Activity>?
}
